Packs/DSPM/Integrations/DSPM/DSPM.py

```diff
@@ -0,0 +1,996 @@
+            "Encrypted": asset.get("encrypted", False),
+            "Open To World": asset.get("openToWorld", False),
+            "Tags": asset.get("tags", {}),
+            "Asset Dig Tags": asset.get("assetDigTags", []),
+        }
+        for asset in assets
+    ]
+    return parsed_assets
+
+
+def get_asset_details(client: Client, args: dict[str, Any]) -> CommandResults:
+    asset_id = args.get("asset_id", None)
+    if not asset_id:
+        raise ValueError("asset_id not specified")
+
+    asset_details = client.get_asset_details(asset_id)
+    demisto.debug(f"Asset details of : {asset_id}")
+    demisto.debug(asset_details)
+
+    headers = ["assetDigTags", "cloudEnvironment", "cloudProvider", "dataTypeGroups", "dataTypes", "encrypted", "id",
+               "lifecycle", "name", "openAlertsCount", "openRisksCount", "openToWorld", "projectId",
+               "projectName", "serviceType", "tags"]
+    readable_output = tableToMarkdown("Asset Details", asset_details, headers=headers, headerTransform=pascalToSpace)
+    return CommandResults(
+        outputs_prefix="DSPM.AssetDetails",
+        outputs_key_field="id",
+        outputs=asset_details,
+        readable_output=readable_output,
+    )
+
+
+def get_asset_files_by_id(client: Client, args: dict[str, Any]) -> CommandResults:
+    asset_id = args.get("asset_id", None)
+    if not asset_id:
+        raise ValueError("Asset ID not specified")
+
+    page_number = 1
+    all_files = []
+
+    while True:
+        params = {
+            "id": asset_id,
+            "page": page_number,
+            "size": MAX_PAGE_SIZE
+        }
+
+        # Fetch the asset files for the current page
+        response = client.get_asset_files(params)
+        files = response.get("files", [])
+
+        if not files:
+            break  # Stop when there are no more files to retrieve
+
+        # Append the files to the total list of files
+        all_files.extend(files)
+
+        # Increment page number for the next fetch
+        page_number += 1
+
+    files_count = len(all_files)
+
+    # Return the result without formatting the files structure
+    headers = ["filename", "path", "type", "size", "openToWorld", "isDeleted", "isMalicious", "dataTypes", "labels", "isDbDump"]
+    readable_output = tableToMarkdown("Asset Files", all_files, headers=headers, headerTransform=pascalToSpace)
+    return CommandResults(
+        outputs_prefix="DSPM.AssetFiles",
+        outputs_key_field="filename",
+        outputs={"files": all_files, "filesCount": files_count},
+        readable_output=readable_output
+    )
+
+
+def get_data_types(client: Client) -> CommandResults:
+    """Command to fetch data types."""
+    data_types = client.get_data_types()
+    data_types_formatted = [
+        {"No": index + 1, "Key": dt} for index, dt in enumerate(data_types)
+    ]
+
+    table_name = "Data Types"
+    headers = ['No', 'Key']
+    if data_types_formatted:
```

Comment:
why we need this condition?

Reply:
We have removed this condition after our discussion.